app/src/main/java/com/canvearth/canvearth/server/FBPixelManager.java

```diff
@@ -101,7 +118,58 @@ public void onCancelled(@NonNull DatabaseError databaseError) {
         return fbPixel;
     }
 
-    public boolean writePixel(PixelData pixelData, Color color, @Nullable Runnable callback) {
+    // You don't have to watch this pixel (for now).. I'm nervous about performance issue of this method.
+    // returns Bitmap which has resolution of 2^resolutionFactor * 2^resolutionFactor
+    // TODO cache this when there is performance issue
+    // Do we need Async version of this?
+    public Bitmap getBitmapSync(PixelData pixelData, int resolutionFactor) {
```

Comment:
This performs a network request & might freeze the main thread. I think we need an Async version.

Reply:
Got it. I will do performance test and decide which way is the best.